src/components/ArticleItem.jsx

```diff
@@ -25,6 +29,41 @@ function ArticleItem() {
         
     }, []);
 
+    const updateArticleVote = (direction, votes) => {
+        let votesInt = parseInt(votes)
+        setArticle(()=> {
+            return {...article, votes: 
+                direction === "increment" 
+                ?
+                votesInt + 1
+                :
+                votesInt - 1
+            
+            }
+        })
+        patchArticleVotes(article_id, direction)
+        .catch((err) => {
+            setErrors((previousState)=> {
+                return [
+                    ...previousState,
+                    {
+                        id: previousState + 1,
```

Comment:
```js
                        id: previousState + 1,
```
This won't work as users can clear errors in whatever order they like. Consider using something like a timestamp (`Date.now()`) or a package like [uuid](https://www.npmjs.com/package/uuid?activeTab=readme) to guarantee that this is unique.